fiftyone/operators/executor.py

```diff
@@ -262,11 +264,26 @@ async def prepare_operator_executor(
     return operator, executor, ctx
 
 
-def _is_generator(value):
-    return isinstance(value, python_types.GeneratorType) or isinstance(
-        value, python_types.AsyncGeneratorType
+async def do_execute_operator(operator, ctx, exhaust=False):
+    result = await (
+        operator.execute(ctx)
+        if asyncio.iscoroutinefunction(operator.execute)
+        else fou.run_sync_task(operator.execute, ctx)
     )
 
+    if not exhaust:
+        return result
+
+    if inspect.isgenerator(result):
```

Comment:
Not a fan of swallowing the messages the operator is sending via the generator. We should probably allow for the callee to pass in a function that processes these. Otherwise there is no path to handling messages when calling `execute_operator()` directly.

Reply:
Although in this could be the behavior of `exhaust` which would be the default. I'm thinking mostlyu about adding `execute_operator(message_reader=MyMessageReader())` or similar.

Reply:
Sounds potentially useful to me 👍 

This PR just unifies how `execute_operator()` work with generators with how delegated operator execution works (which already "swallows" generators)